apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/data_updater/impl.ex

```diff
@@ -2201,7 +2313,27 @@ defmodule Astarte.DataUpdaterPlant.DataUpdater.Impl do
     :ok = TriggersHandler.register_target(trigger_target)
 
     next_device_triggers = Map.put(device_triggers, event_type, new_targets)
+
     Map.put(state, :device_triggers, next_device_triggers)
+    |> maybe_cache_trigger_policy(trigger_target)
+  end
+
+  # TODO: consider what we should to with the cached policy if/when we allow updating a policy
+  defp maybe_cache_trigger_policy(%State{realm: realm_name} = state, %AMQPTriggerTarget{
+         parent_trigger_id: parent_trigger_id
+       }) do
+    case PolicyQueries.retrieve_policy_name(
+           realm_name,
+           parent_trigger_id
+         ) do
+      {:ok, policy_name} ->
+        Map.put(state, :trigger_id_to_policy_name, %{
+          parent_trigger_id => policy_name
+        })
+
+      {:error, :not_found} ->
+        state
```

Comment:
Do we want to ignore this silently or do we want to, e.g., crash if this happens? Is this something that is allowed to happen?

Reply:
This will happen for all triggers which do not specify a policy. As the default policy can't be installed, we simply do not cache it: see [here](https://github.com/Annopaolo/astarte/blob/e05b1557ddb98a5d3b7694c6fd6deed0aa30855a/apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/triggers_handler.ex#L609-L614) (in [the context ](https://github.com/Annopaolo/astarte/blob/e05b1557ddb98a5d3b7694c6fd6deed0aa30855a/apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/triggers_handler.ex#L553) of the `dispatch_event/3` function)